tests/test_data/test_datasets/test_custom_dataset.py

```diff
@@ -23,61 +99,94 @@ def test_custom_classes_override_default(dataset):
-        ann_file=MagicMock(),
+        ann_file=fake_file,
         pipeline=[],
         classes=['bus', 'car'],
         test_mode=True,
-        img_prefix='VOC2007' if dataset == 'VOCDataset' else '')
+        img_prefix='./tests/data/VOCdevkit/VOC2007/' if dataset == 'VOCDataset' else '')
 
     assert custom_dataset.CLASSES != original_classes
     assert custom_dataset.CLASSES == ['bus', 'car']
+    assert repr(custom_dataset) == "bus: 0\tcar: 0" if dataset == 'VOCDataset' else "bus: 4\tcar: 0"
 
     # Test overriding not a subset
     custom_dataset = dataset_class(
-        ann_file=MagicMock(),
+        ann_file=fake_file,
         pipeline=[],
         classes=['foo'],
         test_mode=True,
-        img_prefix='VOC2007' if dataset == 'VOCDataset' else '')
+        img_prefix='./tests/data/VOCdevkit/VOC2007/' if dataset == 'VOCDataset' else '')
 
     assert custom_dataset.CLASSES != original_classes
     assert custom_dataset.CLASSES == ['foo']
+    assert repr(custom_dataset) == "foo: 0" if dataset == 'VOCDataset' else "foo: 4"
 
     # Test default behavior
     custom_dataset = dataset_class(
-        ann_file=MagicMock(),
+        ann_file=fake_file,
         pipeline=[],
         classes=None,
         test_mode=True,
-        img_prefix='VOC2007' if dataset == 'VOCDataset' else '')
+        img_prefix='./tests/data/VOCdevkit/VOC2007/' if dataset == 'VOCDataset' else '')
+
+    if original_classes is None:
+        result = "Dataset is empty"
+    else:
+        instance_count = np.zeros(len(original_classes)).astype(int)
+        if dataset == 'VOCDataset':
+            instance_count[11] = 1
+            instance_count[14] = 1
+        else:
+            instance_count[0] = 4
+        result = ''
+        for cls, count in enumerate(instance_count):
+            result += f"{original_classes[cls]}: {count}"
+            if cls + 1 != len(original_classes):
+                result += '\t'
+
+            if (cls + 1) % 5 == 0:
+                result += '\n'
 
     assert custom_dataset.CLASSES == original_classes
+    assert repr(custom_dataset) == result
 
     # Test sending file path
-    import tempfile
-    tmp_file = tempfile.NamedTemporaryFile()
-    with open(tmp_file.name, 'w') as f:
+    name_file = osp.join(tmp_dir.name, 'fake_name.txt')
+    with open(name_file, 'w') as f:
         f.write('bus\ncar\n')
     custom_dataset = dataset_class(
-        ann_file=MagicMock(),
+        ann_file=fake_file,
         pipeline=[],
-        classes=tmp_file.name,
+        classes=name_file,
         test_mode=True,
-        img_prefix='VOC2007' if dataset == 'VOCDataset' else '')
-    tmp_file.close()
+        img_prefix='./tests/data/VOCdevkit/VOC2007/' if dataset == 'VOCDataset' else '')
+
 
     assert custom_dataset.CLASSES != original_classes
     assert custom_dataset.CLASSES == ['bus', 'car']
+    assert repr(custom_dataset) == "bus: 0\tcar: 0" if dataset == 'VOCDataset' else "bus: 4\tcar: 0"
```

Comment:
The unit test code has been changed a lot and looks a bit messy, is there a more appropriate way to do it

Reply:
I will rewrite this part soon, Thanks